pkg/api-server/inspect_endpoints.go

```diff
@@ -67,6 +70,80 @@ func getMatchedPolicies(
 	}
 }
 
+type gatewayMatchedPolicies struct {
+	listeners []gateway.GatewayListenerInfo
+	proxy     core_xds.Proxy
+}
+
+// getMatchedPoliciesForGateway returns inspection results for each selector of
+// a MeshGateway
+func getMatchedPoliciesForGateway(
+	cfg *kuma_cp.Config, meshContext xds_context.MeshContext, meshGateway *core_mesh.MeshGatewayResource,
+) ([]gatewayMatchedPolicies, error) {
+	proxyBuilder := sync.DefaultDataplaneProxyBuilder(
+		*cfg,
+		callbacks.NewDataplaneMetadataTracker(),
+		envoy.APIV3,
+	)
+
+	// Create a dummy dataplane that serves this MeshGateway
+	dataplaneKey := core_model.ResourceKey{
+		Name: core.NewUUID(),
+		Mesh: meshContext.Resource.Meta.GetName(),
+	}
+
+	var results []gatewayMatchedPolicies
+
+	// There's not really a way of coming up with a canonical fake Dataplane
+	// here, because different selectors could match on different services
+	// so we return multiple results for multiple selectors
+	for _, selector := range meshGateway.Spec.Selectors {
+		dataplane := core_mesh.NewDataplaneResource()
+		dataplane.SetMeta(&rest.ResourceMeta{
+			Type: string(core_mesh.DataplaneType),
+			Name: dataplaneKey.Name,
+			Mesh: dataplaneKey.Mesh,
+		})
+
+		if err := dataplane.SetSpec(&mesh_proto.Dataplane{
+			Networking: &mesh_proto.Dataplane_Networking{
+				Gateway: &mesh_proto.Dataplane_Networking_Gateway{
+					Tags: selector.Match,
```

Comment:
what happens if the selector has `kuma.io/service: *`?

Reply:
Hmm every Gateway selector must have a service tag in fact. Everything works, as far as I can tell at least

Reply:
Could you please try the following patch:
```
Index: pkg/api-server/inspect_endpoints_test.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/pkg/api-server/inspect_endpoints_test.go b/pkg/api-server/inspect_endpoints_test.go
--- a/pkg/api-server/inspect_endpoints_test.go	(revision 0a268cdaa03c235d993ebcc14d932a1426620935)
+++ b/pkg/api-server/inspect_endpoints_test.go	(date 1648720926821)
@@ -486,7 +486,9 @@
 				&core_mesh.TrafficPermissionResource{
 					Meta: &test_model.ResourceMeta{Name: "tp-1", Mesh: "default"},
 					Spec: &mesh_proto.TrafficPermission{
-						Sources:      anyService(),
+						Sources: selectors{
+							serviceSelector("gateway", ""),
+						},
 						Destinations: anyService(),
 					},
 				},
@@ -494,7 +496,7 @@
 					Meta: &test_model.ResourceMeta{Name: "gateway", Mesh: "default"},
 					Spec: &mesh_proto.MeshGateway{
 						Selectors: selectors{
-							serviceSelector("gateway", ""),
+							serviceSelector("*", ""),
 						},
 						Conf: &mesh_proto.MeshGateway_Conf{
 							Listeners: []*mesh_proto.MeshGateway_Listener{

```
I believe result should be the same, but it's different

Reply:
Ah I see, I didn't see the `: *`, good point!